src/preprocess_info.cc

```diff
@@ -158,7 +195,148 @@ Napi::Value PreProcessInfo::GetMeanVariant(const Napi::CallbackInfo& info) {
+  Napi::Object preprocess_channel = Napi::Object::New(env);
+  preprocess_channel.Set("stdScale", preProcessChannel->stdScale);
+  preprocess_channel.Set("meanValue", preProcessChannel->meanValue);
+
+  ie::Blob::Ptr meandata = preProcessChannel->meanData;
+  std::unique_ptr<ie::LockedMemory<void>> locked_memory_;
+  ie::MemoryBlob::Ptr memory_meandata = ie::as<ie::MemoryBlob>(meandata);
+  if (!memory_meandata) {
+    preprocess_channel.Set("meanData",
+                           Napi::ArrayBuffer::ArrayBuffer(env, env.Null()));
+    Napi::Value channelInfo = Napi::Value::From(env, preprocess_channel);
+
+    return channelInfo;
+  }
+
+  locked_memory_.reset(reinterpret_cast<ie::LockedMemory<void>*>(
+      new ie::LockedMemory<const void>(memory_meandata->rmap())));
+  Napi::ArrayBuffer buffer = Napi::ArrayBuffer::New(
+      env,
+      locked_memory_->as<ie::PrecisionTrait<ie::Precision::I8>::value_type*>(),
+      meandata->byteSize());
+
+  preprocess_channel.Set("meanData", buffer);
+  Napi::Value channelInfo = Napi::Value::From(env, preprocess_channel);
+
+  return channelInfo;
+}
+
+void PreProcessInfo::SetPreProcessChannel(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+
+  if (info.Length() != 2) {
+    Napi::TypeError::New(env, "Wrong number of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  if (!info[0].IsNumber()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  if (!info[1].IsObject()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  size_t index = info[0].ToNumber().Int32Value();
+  Napi::Object newChannel = info[1].ToObject();
+
+  ie::PreProcessInfo& preInfo = _input_info->getPreProcess();
+
+  size_t numberOfChannels = preInfo.getNumberOfChannels();
+
+  if (numberOfChannels == 0) {
+    Napi::Error::New(env, "accessing pre-process when nothing was set.")
+        .ThrowAsJavaScriptException();
+    return;
+  };
+
+  if (index >= numberOfChannels) {
+    std::string errorLog =
+        "pre process index " + std::to_string(index) + " is out of bounds.";
+    Napi::Error::New(env, errorLog).ThrowAsJavaScriptException();
+    return;
+  };
+
+  ie::PreProcessChannel::Ptr channel = preInfo[index];
+  if (newChannel.Has("stdScale")) {
+    channel->stdScale = newChannel.Get("stdScale").ToNumber().FloatValue();
+  }
+
+  if (newChannel.Has("meanValue")) {
+    channel->meanValue = newChannel.Get("meanValue").ToNumber().FloatValue();
+  }
+
+  if (newChannel.Has("meanData")) {
+    Napi::ArrayBuffer meanDataArray =
+        Napi::ArrayBuffer::ArrayBuffer(env, newChannel.Get("meanData"));
+    size_t bytelength = meanDataArray.ByteLength();
+    void* buffer = meanDataArray.Data();
+
+    try {
+      ie::SizeVector dims_vector = {1, bytelength, 1, 1};
+      ie::TensorDesc tensor(ie::Precision::I8, dims_vector, ie::Layout::NCHW);
```

Comment:
The precision should be `ie::Precision::FP32`. The layout should be `ie::Layout::HW`.

Reply:
Done.